modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableRowSkinBase.java

```diff
@@ -542,18 +542,18 @@ void updateCells(boolean resetChildren) {
 
         // update children of each row
         if (fixedCellSizeEnabled) {
-            // we leave the adding / removing up to the layoutChildren method mostly,
-            // but here we remove any children cells that refer to columns that are
-            // not visible
+            // we leave the adding / removing up to the layoutChildren method mostly, but here we remove any children
+            // cells that refer to columns that are removed or not visible.
             List<Node> toRemove = new ArrayList<>();
             for (Node cell : getChildren()) {
-                if (! (cell instanceof IndexedCell)) continue;
-                if (!getTableColumn((R)cell).isVisible()) {
+                if (!(cell instanceof IndexedCell)) continue;
+                TableColumnBase<T, ?> tableColumn = getTableColumn((R) cell);
+                if (!getVisibleLeafColumns().contains(tableColumn) || !tableColumn.isVisible()) {
                     toRemove.add(cell);
                 }
             }
             getChildren().removeAll(toRemove);
-        } else if (!fixedCellSizeEnabled && (resetChildren || cellsEmpty)) {
+        } else if (resetChildren || cellsEmpty) {
```

Comment:
just curious : any idea why fixedCellSize was special-cased here? Not to clean them up sounds (and is) very much wrong, so there must have been a reason?

Reply:
The '!fixedCellSizeEnabled' check is not needed, as we already check for fixedCellSizeEnabled in the main if condition.
`if (fixedCellSizeEnabled) {...}`

So before, it was like: 
`if (fixedCellSizeEnabled) {...} `
`else if (!fixedCellSizeEnabled && (resetChildren || cellsEmpty)) {...}` 

So we only execute the else condition, if fixedCellSizeEnabled is not true -> !fixedCellSizeEnabled. So the check is not necessary, as fixedCellSizeEnabled must be false, when we come to the else condition.
And fixedCellSizeEnabled is also a primitive, so it can't be null, making this check obsolete.

Reply:
yeah, was curious about the stuff in your "edit" paragraph. And thanks for the explanation of the if-block, didn't see the whole .. :)